servicetalk-encoding-netty/src/main/java/io/servicetalk/encoding/netty/NettyChannelContentCodec.java

```diff
@@ -0,0 +1,342 @@
+                channel.writeInbound(toByteBuf(src));
+                Buffer buffer = drainChannelQueueToSingleBuffer(channel.inboundMessages(), allocator);
+
+                if (buffer != null && buffer.readableBytes() > 0) {
+                    subscriber.onNext(buffer);
+                } else {
+                    // Not enough data to decompress, ask for more
+                    subscription.request(1);
+                }
+            }
+
+            @Override
+            public void onError(final Throwable t) {
+                safeCleanup(channel);
+                subscriber.onError(t);
+            }
+
+            @Override
+            public void onComplete() {
+                try {
+                    cleanup(channel);
+                } catch (Throwable t) {
+                    subscriber.onError(t);
+                    return;
+                }
+
+                subscriber.onComplete();
+            }
+        });
+    }
+
+    private EmbeddedChannel newEmbeddedChannel(final ChannelHandler handler, final BufferAllocator allocator) {
+        final EmbeddedChannel channel = new EmbeddedChannel(handler);
+        channel.config().setAllocator(getByteBufAllocator(allocator));
+        return channel;
+    }
+
+    private static ByteBuf toByteBuf(final Buffer buffer) {
+        return BufferUtils.extractByteBufOrCreate(buffer);
+    }
+
+    private static ByteBuf toByteBuf(final Buffer buffer, final int offset, final int length) {
+        return BufferUtils.extractByteBufOrCreate(buffer.readerIndex(buffer.readerIndex() + offset).readSlice(length));
+    }
+
+    private static void preparePendingData(final EmbeddedChannel channel) {
+        try {
+            channel.close().syncUninterruptibly().get();
+            channel.checkException();
+        } catch (InterruptedException | ExecutionException ex) {
+            throwException(ex);
+        }
+    }
+
+    @Nullable
+    private static Buffer drainChannelQueueToSingleBuffer(
+            final Queue<Object> queue, final BufferAllocator allocator) {
+        if (queue.isEmpty()) {
+            return null;
+        }
+
+        if (queue.size() == 1) {
+            return newBufferFrom((ByteBuf) queue.poll());
+        } else {
+            int accumulateSize = 0;
+
+            List<Buffer> parts = new ArrayList<>(queue.size());
+            ByteBuf part;
+            while ((part = (ByteBuf) queue.poll()) != null) {
+                parts.add(newBufferFrom(part));
+                accumulateSize += part.readableBytes();
+            }
+
+            // Try to merge everything together if total size is less than 1KiB (small chunks, ie. footer).
+            // CompositeBuffer can requires some additional work to write and may limit how much data we can
+            // pass to writev (max of 1024 pointers), so if there are small chunks it can be better to combine them.
+            if (accumulateSize <= 1 << 10) {
```

Comment:
Consider using a constant and just use 1024 to improve readability of the code. 1024 is a simple value, it takes longer to compute `1 << 10` in the head :)